core/src/main/java/com/linecorp/armeria/client/ClientFactory.java

```diff
@@ -369,4 +377,14 @@ default ClientBuilderParams validateParams(ClientBuilderParams params) {
         }
         return params;
     }
+
+    /**
+     * Registers a JVM shutdown hook that closes this {@link ClientFactory} when the current JVM terminates.
+     */
+    default void closeOnShutdown() {
+        Runtime.getRuntime().addShutdownHook(THREAD_FACTORY.newThread(() -> {
+            closeDefault();
+            logger.debug("ClientFactory has been closed.");
+        }));
+    }
```

Comment:
Should we implement this method in `DefaultClientFactory` rather than implementing here so that we can hide `THREAD_FACTORY` and `logger` from `public`?

Reply:
I meant to make this abstract method and implement this method in `DefaultClientFactory`. 😅
```java
interface ClientFactory {
    void closeOnShutdown();
    void closeOnShutdown(@Nullable Runnable whenClosing);

}
```

Reply:
Gentle ping. This review is not addressed yet. 😉